### PR TITLE
Closes #275

### LINKED ISSUE #275: Rules methods cleanup: NextBestMinDist
Please edit appropriately below if needed.

To do:

- [ ] Revise documentation
  - [ ] RStudio outline section dividers (use `# function or class name ----`)
  - [ ] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [ ] Replace `##'` by `#'`
  - [ ] Add to `pkgdown.yml`
  - [ ] Remove file history and the timestamp in comments on top of file
- [ ] Add unit tests
  - [ ] For classes:
    - [ ] Test default constructor
    - [ ] For reference classes: test methods
  - [ ] For functions: as usual
- [ ] Refactor as needed
  - [ ] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [ ] Keep externally used names so that we don't break downstream code
  - [ ] Remove `keywords` tags
  - [ ] For functions or methods:
    - [ ] Add assertions for arguments using `checkmate` functions
    - [ ] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [ ] Consider renaming internal variables etc. if needed
    - [ ] Avoid explicit use of `return`
  - [ ] For classes:
    - [ ] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [ ] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [ ] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #275

### CODE REVIEW

File: R/Rules-class.R [hunk 1]
```diff
@@ -371,7 +371,7 @@ NextBestDualEndpoint <- function(target,
 #' [`NextBestMinDist`] is the class for next best dose that is based on minimum
 #' distance to target probability.
 #'
-#' @slot target (`proportion`)\cr target toxicity probability.
+#' @slot target (`proportion`)\cr target toxicity probability, a scalar value.
```
Comment (danielinteractive) [inline R/Rules-class.R:374]: maybe easier to read as:
```suggestion
#' @slot target (`proportion`)\cr single target toxicity probability.
```
?

**Review by danielinteractive — APPROVED** [review]
thanks Wojciech, only minor comments inside, after addressing those good to merge. I approve now so that you don't need to wait for me. thx

File: R/Rules-class.R (hunk 1 above)
Comment (wwojciech) [reply R/Rules-class.R:374]: sure!

File: R/Rules-methods.R [hunk 2]
```diff
@@ -650,34 +650,40 @@ setMethod(
...
   ),
-  def = function(nextBest, doselimit, samples, model, data, ...) {
-    dosesOK <-
-      if (length(doselimit)) {
-        which(data@doseGrid <= doselimit)
-      } else {
-        seq_along(data@doseGrid)
-      }
+  definition = function(nextBest, doselimit, samples, model, data, ...) {
+    doselimit <- ifelse(missing(doselimit) || length(doselimit) == 0, Inf, doselimit)
```
[the remark this answers is not in the document]
Comment (wwojciech) [reply R/Rules-methods.R:672]: Yap, this is actually a workaround until this #261 issue is solved. This issue is to specify a default value for `doselimit`, i.e. `doselimit = Inf`. After it is solved, this line will be delated. But I am waiting to do this after all the `nextBest` methods will be cleaned-up.

File: R/Rules-methods.R [hunk 3]
```diff
@@ -650,34 +650,40 @@ setMethod(
...
-    probDLT <- modelfit$middle[dosesOK]
-    doses <- modelfit$dose[dosesOK]
-    bestIndex <- which.min(abs(probDLT - nextBest@target))
-    bestDose <- doses[bestIndex]
-    return(list(value = bestDose))
+    doses <- modelfit$dose
+    is_dose_eligible <- doses <= doselimit
+    dlt_prob <- modelfit$middle[is_dose_eligible]
+    next_best_level <- which.min(abs(dlt_prob - nextBest@target))
+    next_best <- doses[is_dose_eligible][next_best_level]
```
[the remark this answers is not in the document]
Comment (wwojciech) [reply R/Rules-methods.R:679]: sure!

**Pull request merged**